kahuna/public/js/components/gr-archiver/gr-archiver.js

```diff
@@ -0,0 +1,62 @@
+import angular from 'angular';
+import Rx from 'rx';
+
+import './gr-archiver.css!';
+import template from './gr-archiver.html!text';
+
+import '../../services/archive';
+import '../../services/image-accessor';
+
+
+export const module = angular.module('gr.archiver', [
+    'kahuna.services.archive',
+    'kahuna.services.image-accessor'
+]);
+
+module.controller('grArchiverCtrl', [
+    '$scope',
+    'archiveService',
+    'imageAccessor',
+    function ($scope,
+              archiveService,
+              imageAccessor) {
+
+        const ctrl = this;
+
+        $scope.$watchCollection(() => ctrl.images.toJS(), (images) => {
+            const allArchived = images.every(imageAccessor.isArchived);
+            ctrl.archivedState = allArchived ? 'archived' : 'unarchived';
```

Comment:
Technically this watch gets executed immediately when the controller is initialised, but I imagine you're referring to making it obvious what properties exist on that controller?

I don't really like repetition but I think explicitness may be worth it in this case. Could do the same with `ctrl.archiving = false`.